src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,205 @@
+	const [authorToBeCreated, setAuthorToBeCreated] = useState('');
+	const [_authorsList, _setAuthorsList] = useState(authorsList);
+
+	function handleAddAuthor(e, authorToBeAdded) {
+		e.preventDefault();
+		for (let author of courseAuthorsList)
+			if (author.id === authorToBeAdded.id) return;
+		setCourseAuthorsList([
+			...courseAuthorsList,
+			{ id: authorToBeAdded.id, name: authorToBeAdded.name },
+		]);
+		_setAuthorsList(
+			_authorsList.filter((author) => author.id !== authorToBeAdded.id)
+		);
+	}
+
+	function handleDeleteAuthor(e, authorToBeDeleted) {
+		e.preventDefault();
+		for (let author of courseAuthorsList)
+			if (author.id === authorToBeDeleted.id) {
+				setCourseAuthorsList(
+					courseAuthorsList.filter(
+						(author) => author.id !== authorToBeDeleted.id
+					)
+				);
+				_setAuthorsList([
+					..._authorsList,
+					{ id: authorToBeDeleted.id, name: authorToBeDeleted.name },
+				]);
+			}
+	}
+
+	function handleCreateAuthor(e) {
+		e.preventDefault();
+		if (!authorToBeCreated) return;
+		setAuthorsList([
+			...authorsList,
+			{
+				id: uuid(),
+				name: authorToBeCreated,
+			},
+		]);
+		_setAuthorsList([
+			..._authorsList,
+			{
+				id: uuid(),
+				name: authorToBeCreated,
+			},
+		]);
+	}
+
+	function handleCreateCourse(e) {
+		e.preventDefault();
+		if (courseAuthorsList.length === 0) {
+			alert('Authors need to be added');
+			return;
+		}
+		if (courseDuration === 0) {
+			alert('Course duration must be greater than 0');
+			return;
+		}
+		let creationDate = new Date().toLocaleDateString().replace('/', '.');
+		setCoursesList([
+			...coursesList,
+			{
+				id: uuid(),
+				title: courseTitle,
+				description: courseDescription,
+				creationDate: creationDate,
+				duration: courseDuration,
+				authors: courseAuthorsList.map((author) => author.id),
+			},
+		]);
+		setCreateCourseMode(false);
+	}
+
+	return (
+		<>
+			<form onSubmit={handleCreateCourse}>
+				<div className='title-and-create'>
+					<Input
+						id='course-title-input'
+						className='course-title'
+						value={courseTitle}
+						onInput={(e) => setCourseTitle(e.target.value)}
+						placeholder='Enter title...'
+						labelClassName='course-title-input-label'
+						labelText='Title'
+						required
+					/>
+					<div className='create-course'>
+						<Button
+							buttonText='Create course'
+							className='create-course-button'
+						/>
+					</div>
+				</div>
+				<div className='description'>
+					<label htmlFor='course-description' className='description-label'>
+						Description
+					</label>
+					<textarea
+						id='course-description'
+						className='description-textarea'
+						placeholder='Enter description...'
+						value={courseDescription}
+						onInput={(e) => setCourseDescription(e.target.value)}
+						required
+					/>
+				</div>
+				<div className='course-info'>
+					<div className='create-author-and-duration'>
+						<h2>Add author</h2>
+						<Input
```

Comment:
Author field (input) - author name length should be at least 2 characters;

Reply:
Done!